Should limit console output of unit test

- [x] I have searched the [issues](https://github.com/apache/incubator-dubbo/issues) of this repository and believe that this is not a duplicate.
- [x] I have checked the [FAQ](https://github.com/apache/incubator-dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: master branch
* Operating System version: xxx
* Java version: xxx

### Steps to reproduce this issue

Travis-cli has a 4MB hard limit of log size [1].
Currently, our Travis log[2] is about 3.98MB, we are almost there.
We can reduce some trivial log in the console output, but keep a full log in the local file.

For example, test case `org.apache.dubbo.rpc.validation.ValidationTest` eats about 1.3MB in travis log. We can start from here.

[1]. https://docs.travis-ci.com/user/common-build-problems/#log-length-exceeded
[2]. https://api.travis-ci.org/v3/job/484750418/log.txt

Pls. provide [GitHub address] to reproduce this issue.

### Expected Result

What do you expected from the above steps？

### Actual Result

What actually happens?

If there is an exception, please attach the exception trace:

```
Just put your stack trace here!
```

#### for Travis environment
We can output all the unit test log to one local file maybe `PROJECT_ROOT/dubbo.log`.
If there is some error, use Travis `after_failure` to print last information of `dubbo.log`.
```
after_failure:
  - if [ -f dubbo.log ]; then echo "------TAIL of dubbo.log------"; tail -n 1000 dubbo.log; echo "------END of dubbo.log------"; fi
```

I simulated a test failure, and the Travis log is https://api.travis-ci.org/v3/job/485746697/log.txt
```
travis_time:end:238f76ac:start=1548741013072176158,finish=1548741024298622064,duration=11226445906
[0Ktravis_fold:end:cache.2
[0K
travis_fold:start:after_failure
[0Ktravis_time:start:2672b980
[0K$ if [ -f dubbo.log ]; then echo "------TAIL of dubbo.log------"; tail -n 1000 dubbo.log; echo "------END of dubbo.log------";fi
------TAIL of dubbo.log------
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:582)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:499)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:461)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:884)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: TestService
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)

// SOME STACK INFO HERE

	... 27 more
2019-01-29 05:50:12,177 WARN [org.apache.dubbo.remoting.transport.AbstractServer:disconnected] -  [DUBBO] All clients has disconnected from /172.17.0.1:34644. You can graceful shutdown now., dubbo version: 2.7.1-SNAPSHOT, current host: 172.17.0.1
------END of dubbo.log------
travis_time:end:2672b980:start=1548741024303108412,finish=1548741024308579986,duration=5471574
[0Ktravis_fold:end:after_failure
[0K
Done. Your build exited with 1.
```

#### for local environment
We can always check the full content of the local file log.

If everything is fine, the Travis log will be like https://api.travis-ci.org/v3/job/485755453/log.txt
And the full view of the commit https://github.com/biyuhao/incubator-dubbo/commit/1a103d1e8c956e8981aa415bde1710a3f0063011

--- a/.travis.yml
+++ b/.travis.yml
@@ -16,3 +16,6 @@ script:
 
 after_success:
   - bash <(curl -s https://codecov.io/bash)
+
+after_failure:
+  - if [ -f dubbo.log ]; then echo "------TAIL of dubbo.log------"; tail -n 1000 dubbo.log; echo "------END of dubbo.log------"; fi

--- a/dubbo-cluster/src/test/resources/log4j.xml
+++ b/dubbo-cluster/src/test/resources/log4j.xml
@@ -18,6 +18,7 @@
 <!DOCTYPE log4j:configuration SYSTEM "log4j.dtd">
 <log4j:configuration xmlns:log4j="http://jakarta.apache.org/log4j/" debug="false">
     <appender name="DUBBO" class="org.apache.dubbo.common.utils.DubboAppender">
+        <param name="File" value="../dubbo.log"/>
         <layout class="org.apache.log4j.PatternLayout">
             <param name="ConversionPattern" value="[%d{dd/MM/yy HH:mm:ss:SSS z}] %t %5p %c{2}: %m%n"/>
         </layout>

--- a/dubbo-common/src/main/java/org/apache/dubbo/common/utils/DubboAppender.java
+++ b/dubbo-common/src/main/java/org/apache/dubbo/common/utils/DubboAppender.java
@@ -16,13 +16,18 @@
  */
 package org.apache.dubbo.common.utils;
 
-import org.apache.log4j.ConsoleAppender;
+import org.apache.log4j.FileAppender;
 import org.apache.log4j.spi.LoggingEvent;
 
 import java.util.ArrayList;
 import java.util.List;
 
-public class DubboAppender extends ConsoleAppender {
+public class DubboAppender extends FileAppender {
+
+    public DubboAppender() {
+        super();
+        setFile("dubbo.log");
+    }
 
     public static boolean available = false;
 

--- a/dubbo-common/src/main/java/org/apache/dubbo/common/utils/DubboAppender.java
+++ b/dubbo-common/src/main/java/org/apache/dubbo/common/utils/DubboAppender.java
@@ -24,14 +24,16 @@ import java.util.List;
 
 public class DubboAppender extends FileAppender {
 
+    private static final String DEFAULT_FILE_NAME = "dubbo.log";
+
     public DubboAppender() {
         super();
-        setFile("dubbo.log");
+        setFile(DEFAULT_FILE_NAME);
     }
 
     public static boolean available = false;
 
-    public static List<Log> logList = new ArrayList<Log>();
+    public static List<Log> logList = new ArrayList<>();
 
     public static void doStart() {
         available = true;

--- a/dubbo-common/src/test/resources/log4j.xml
+++ b/dubbo-common/src/test/resources/log4j.xml
@@ -20,6 +20,7 @@
     <!-- 以下是appender的定义 -->
     <!-- ===================================================================== -->
     <appender name="dubbo" class="org.apache.dubbo.common.utils.DubboAppender">
+        <param name="File" value="../dubbo.log"/>
         <param name="encoding" value="GBK"/>
         <layout class="org.apache.log4j.PatternLayout">
             <param name="ConversionPattern" value="%d %p [%c:%M] - %m%n"/>

--- a/dubbo-registry/dubbo-registry-default/src/test/resources/log4j.xml
+++ b/dubbo-registry/dubbo-registry-default/src/test/resources/log4j.xml
@@ -21,6 +21,7 @@
     <!-- The following is the definition of appender -->
     <!-- ===================================================================== -->
     <appender name="dubbo" class="org.apache.dubbo.common.utils.DubboAppender">
+        <param name="File" value="../../dubbo.log"/>
         <param name="encoding" value="GBK"/>
         <layout class="org.apache.log4j.PatternLayout">
             <param name="ConversionPattern" value="%d %p [%c:%M] - %m%n"/>
@@ -30,14 +31,8 @@
             <param name="LevelMax" value="DEBUG" />
         </filter> -->
     </appender>
-    <appender name="CONSOLE" class="org.apache.log4j.ConsoleAppender">
-        <layout class="org.apache.log4j.PatternLayout">
-            <param name="ConversionPattern" value="[%d{dd/MM/yy HH:mm:ss:SSS z}] %5p %c{2}: %m%n"/>
-        </layout>
-    </appender>
     <root>
         <level value="INFO"/>
         <appender-ref ref="dubbo"/>
-        <appender-ref ref="CONSOLE"/>
     </root>
 </log4j:configuration>

--- a/dubbo-remoting/dubbo-remoting-netty/src/test/resources/log4j.xml
+++ b/dubbo-remoting/dubbo-remoting-netty/src/test/resources/log4j.xml
@@ -21,6 +21,7 @@
     <!-- 以下是appender的定义 -->
     <!-- ===================================================================== -->
     <appender name="dubbo" class="org.apache.dubbo.common.utils.DubboAppender">
+        <param name="File" value="../../dubbo.log"/>
         <param name="encoding" value="GBK"/>
         <layout class="org.apache.log4j.PatternLayout">
             <param name="ConversionPattern" value="%d %p [%c:%M] - %m%n"/>

--- a/dubbo-rpc/dubbo-rpc-api/src/test/resources/log4j.xml
+++ b/dubbo-rpc/dubbo-rpc-api/src/test/resources/log4j.xml
@@ -21,6 +21,7 @@
     <!-- 以下是appender的定义 -->
     <!-- ===================================================================== -->
     <appender name="dubbo" class="org.apache.dubbo.common.utils.DubboAppender">
+        <param name="File" value="../../dubbo.log"/>
         <param name="encoding" value="GBK"/>
         <layout class="org.apache.log4j.PatternLayout">
             <param name="ConversionPattern" value="%d %p [%c:%M] - %m%n"/>

--- a/dubbo-rpc/dubbo-rpc-dubbo/src/test/resources/log4j.xml
+++ b/dubbo-rpc/dubbo-rpc-dubbo/src/test/resources/log4j.xml
@@ -21,6 +21,7 @@
     <!-- 以下是appender的定义 -->
     <!-- ===================================================================== -->
     <appender name="dubbo" class="org.apache.dubbo.common.utils.DubboAppender">
+        <param name="File" value="../../dubbo.log"/>
         <param name="encoding" value="GBK"/>
         <layout class="org.apache.log4j.PatternLayout">
             <param name="ConversionPattern" value="%d %p [%c:%M] - %m%n"/>
@@ -30,17 +31,8 @@
             <param name="LevelMax" value="DEBUG" />
         </filter> -->
     </appender>
-    <appender name="FILE" class="org.apache.log4j.FileAppender">
-        <param name="File" value="dubbo.log"/>
-        <layout class="org.apache.log4j.PatternLayout">
-            <!-- <param name="ConversionPattern" value="[%t %d{dd/MM/yy HH:mm:ss:SSS
-                z}] %5p %c{2}: %L %m%n" /> -->
-            <param name="ConversionPattern" value="[%t %l %d{dd/MM/yy HH:mm:ss:SSS z}] %5p %m %n"/>
-        </layout>
-    </appender>
     <root>
         <level value="INFO"/>
         <appender-ref ref="dubbo"/>
-        <appender-ref ref="FILE"/>
     </root>
 </log4j:configuration>

--- a/dubbo-serialization/dubbo-serialization-test/src/test/resources/log4j.xml
+++ b/dubbo-serialization/dubbo-serialization-test/src/test/resources/log4j.xml
@@ -18,6 +18,7 @@
 <!DOCTYPE log4j:configuration SYSTEM "log4j.dtd">
 <log4j:configuration xmlns:log4j="http://jakarta.apache.org/log4j/">
     <appender name="dubbo" class="org.apache.dubbo.common.utils.DubboAppender">
+        <param name="File" value="../../dubbo.log"/>
         <param name="encoding" value="GBK"/>
         <layout class="org.apache.log4j.PatternLayout">
             <param name="ConversionPattern" value="%d %p [%c:%M] - %m%n"/>

